starlite/new_dto/__init__.py

```diff
@@ -0,0 +1,5 @@
+from __future__ import annotations
```

Comment:
Why do you have a `new_dto` module? 

This should just be `dto`, please remove the old DTO stuff while at it. 

We either have this as a feature branch, not on main, or we have main in a partial state waiting for the follow-up - I'm fine with either. 

I'm not fine though with a mongrel state.

Reply:
I just wanted to keep it isolated from the other stuff while I iterated over it.

Always intended it to become `dto`, but we'd never explicitly spoke about deleting the old implementation - I will do that.

Reply:
>  or we have main in a partial state waiting for the follow-up - I'm fine with either.

This wouldn't be a good solution IMO. If we're to do another alpha release or something like that, this is going to interfere with it. Also, conceptually, merging a PR to main should not leave main in a partial state.

@peterschutt If you want to tackle this in increments, how about a feature branch? Then we could merge that into `main` once it's done.

Reply:
Yeh good point. I'll leave this branch as the feature branch then.

Reply:
Sounds good. Just have followup PRs pointing at this one